internal/adminapi/endpoints.go

```diff
@@ -0,0 +1,76 @@
+package adminapi
+
+import (
+	"context"
+	"fmt"
+
+	discoveryv1 "k8s.io/api/discovery/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetURLsForService performs an endpoint lookup, using provided kubeClient
+// to list provided service's endpointslices.
+func GetURLsForService(ctx context.Context, kubeClient client.Client, service types.NamespacedName) ([]string, error) {
+	// Get all the endpointslices assigned to the provided service.
+	labelReq, err := labels.NewRequirement("kubernetes.io/service-name", selection.Equals, []string{service.Name})
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		addresses     []string
+		continueToken string
+	)
+	for {
+		var endpointsList discoveryv1.EndpointSliceList
+		if err := kubeClient.List(ctx, &endpointsList, &client.ListOptions{
+			LabelSelector: labels.NewSelector().Add(*labelReq),
+			Namespace:     service.Namespace,
+			Continue:      continueToken,
```

Comment:
What the Continue is needed for without having a Limit set? Isn't it used to get incremental chunks of elements instead the whole set at once?